go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -2173,3 +2186,86 @@ func generateIPFragmentReassemblyFlow(ofPortPhys string) []string {
 
 	return flows
 }
+
+// deleteStaleMasqueradeResources() function removes stale Linux resources when
+// config.Gateway.V4MasqueradeSubnet or config.Gateway.V4MasqueradeSubnet gets
+// changed at day 2.
+func deleteStaleMasqueradeResources(bridgeName string) error {
+	link, err := util.LinkSetUp(bridgeName)
+	if err != nil {
+		return fmt.Errorf("unable to get link for %s, error: %v", bridgeName, err)
+	}
+
+	// Determine old masquerade subnet by listing all address for a bridge and
+	// then by filtering using label, ipfamily and current masquerade subnet.
+	// Label gets added to the bridge while adding V4HostMasqueradeIP or V6HostMasqueradeIP
+	// address previously through setNodeMasqueradeIPOnExtBridge() function.
+	addrs, err := util.GetNetLinkOps().AddrList(link, netlink.FAMILY_ALL)
+	if err != nil {
+		return fmt.Errorf("failed to list addresses for the link %s: %v", link.Attrs().Name, err)
+	}
+
+	var staleMasqueradeIPs config.MasqueradeIPsConfig
+	for _, addr := range addrs {
+		if addr.Label == fmt.Sprintf("%s-masq", link.Attrs().Name) && !utilnet.IsIPv6(addr.IP) &&
+			!addr.IPNet.Contains(config.Gateway.MasqueradeIPs.V4OVNMasqueradeIP) {
```

Comment:
Why `Contains` instead of direct address comparison?

Reply:
yaa, makes sense to check equal to V4HostMasqueradeIP instead of Contains.